scripts/Diffraction/isis_powder/routines/focus.py

```diff
@@ -66,23 +63,24 @@ def _focus_one_ws(input_workspace, run_number, instrument, perform_vanadium_norm
         if expected_filename in os.listdir(save_directory):
             input_workspace = mantid.LoadNexus(os.path.join(save_directory, expected_filename))
         else:
-            corrections = mantid.PaalmanPingsMonteCarloAbsorption(
-                InputWorkspace=input_workspace,
-                Shape=sample_details._shape_type.capitalize(),
-                BeamHeight=2.0,
-                BeamWidth=2.0,
-                Height=sample_details.height(),
-                SampleRadius=sample_details._shape.radius,
-                SampleChemicalFormula=sample_details.material_object.chemical_formula,
-                ContainerRadius=container_details.radius,
-                ContainerChemicalFormula=container_details.material_object.chemical_formula,
-                EventsPerPoint=events_per_point
-            )
+            mantid.SetSample(input_workspace, Geometry={'Shape': sample_details._shape_type.capitalize(),
```

Comment:
Please can you see if you can add a method to the sample_details class called something like `get_sample_geometry` that generates this json string? I think that will make it easier to support the slab shape as well as the cylinder
This function will essentially be the same as `generate_sample_geometry` in common.py but I think it should be part of the sample_details class. As I mentioned in our call I think `generate_sample_geometry` should be removed once you've done this